external/external.sh

```diff
@@ -0,0 +1,134 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# script runs in 3 modes - build / run / clean
+
+set -e
+tag=${DOCKERIMAGE_TAG}
+docker_os=ubuntu
+build_type=full
+package=jdk
+command_type=build
+version=8
+impl=hotspot
+test=derby
+report="false"
+docker_args="--rm"
+
+usage () {
+	echo 'Usage : external.sh  --dir TESTDIR --tag DOCKERIMAGE_TAG --version JDK_VERSION --impl JDK_IMPL [--report REPORTDIR] [--docker_args EXTRA_DOCKER_ARGS] [--build|--run|--clean]'
+}
+
+parseCommandLineArgs()
+{
+	while [[ $# -gt 0 ]] && [[ ."$1" = .-* ]] ; do
+		opt="$1";
+		shift; 
+
+		case "$opt" in
+			"--dir" | "-d" )
+				test="$1"; shift;;
+			
+			"--version" | "-v" )
+				version="$1"; shift;;
+			
+			"--impl" | "-i" )
+				impl="$1"; shift;;
+
+			"--docker_args" )
+				if [ -z ${1+x} ]; then 
+					echo "No EXTRA_DOCKER_ARGS set"; 
+				else 
+  					docker_args="--rm $1"; shift;
+				fi;;
+				
+			"--tag" | "-t" )
+				tag="$1"; shift; parse_tag;;
+
+			"--report" )
+				report="$1"; shift;;
+
+			"--build" | "-b" )
+				command_type=build;; 
+
+			"--run" | "-r" )
+				command_type=run;;
+			
+			"--clean" | "-c" )
+				command_type=clean;;
+
+			"--help" | "-h" )
+				usage; exit 0;;
+
+			*) echo >&2 "Invalid option: ${opt}"; echo "This option was unrecognized."; usage; exit 1;
+		esac
+	done	
+}
+
+# Parse environment variable DOCKERIMAGE_TAG
+# to set docker_os, build_type, package
+function parse_tag () { 
+
+	# set PACKAGE
+	case $tag in
+   		*jre*) 
+    		package=jre
+		;;
+	esac
+	# set BUILD_TYPE
+	case $tag in
+   		*-slim*|*_slim*) 
+   			build_type=slim
+   		;;
+	esac
+	# set DOCKER_OS
+	case $tag in
+   		*alpine*) 
+	   		docker_os=alpine;;
+   		*debianslim*) 
+	   		docker_os=debianslim;;
+		*debian*) 
+	   		docker_os=debian;;
+		*centos*) 
+	   		docker_os=centos;;
+		*clefos*) 
+	   		docker_os=clefos;;
+		*ubi-minimal*) 
+	   		docker_os=ubi-minimal;;
+		*ubi*) 
+	   		docker_os=ubi;;
+		*ubuntu*|*latest*|*nightly*) 
+	   		docker_os=ubuntu;;
+   		*) echo "Unable to recognize DOCKER_OS from DOCKERIMAGE_TAG = $tag!";;
+	esac     
+}
+
+parseCommandLineArgs "$@"
+
+if [ $command_type == "build" ]; then
+	source $(dirname "$0")/build_image.sh $test $version $impl $docker_os $package $build_type
+fi
+
+if [ $command_type == "run" ]; then
+	docker run $docker_args adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type
+fi
+
+if [ $report != "false" ]; then
+	docker cp adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type:/testResults/surefire-reports $report/external_test_reports;
+fi
+
+if [ $command_type == "clean" ]; then
+	docker rmi -f adoptopenjdk-$test-test:${JDK_VERSION}-$package-$docker_os-${JDK_IMPL}-$build_type
```

Comment:
Might be better to also include removing containers.

Reply:
@sophia-guo Are you talking about removing the base containers? Like `adoptopenjdk:openjdk13-openj9`? Cause the above command will remove the test container built upon the `adoptopenjdk`, but not the base container itself.

Reply:
Sorry, I didn't realize that we actually using --rm to delete anonymous test containers. That is being said all is good.

Reply:
ok sweet

Reply:
Currently there are two different scenarios. For tests do not need copy the results we can use --rm to automatically remove the container when it exits. For tests need to copy the test results we can't remove the container when exists. Instead we assign the container name when do `docker run --name`.  And  the container name will be used when doing `docker cp containername:directory....`.  In this scenario when doing clean the test container should also be removed. Sorry I just realized this.

Reply:
ah that explains it.  Let me attempt to generalize this enough to make it possible.  Essentially I would like all test results from all apps made available... but I recognize some may not be in obvious locations